internal/pkg/agent/install/uninstall.go

```diff
@@ -37,6 +37,14 @@ import (
 
 // Uninstall uninstalls persistently Elastic Agent on the system.
 func Uninstall(cfgFile, topPath, uninstallToken string, log *logp.Logger, pt *progressbar.ProgressBar) error {
+	// Immediately fail it tamper protection is enabled but no uninstallToken is specified
+	if features.TamperProtection() && uninstallToken == "" {
```

Comment:
I may be misunderstanding what the `features` package provides; currently if we enroll an agent into a policy without tamper protection enabled this call returns true. @aleksmaus is this intended behaviour?

Reply:
This was the "feature flag" initially. I think is was disabled by default in the initial release, and I think it was enabled by default in subsequent releases.